### DISCUSSION TITLE
paddleocr with pyinstaller

### DISCUSSION THREAD

**Discussion by WENDGOUNDI** [PaddlePaddle/PaddleOCR#6875] [General]
Hi everyone. Is anyone already created an executable file using pyinstaller with paddleocr in his python file as ocr engine? I have been trying the process for few weeks now without satisfying result. Thanks in advance
![paddleocr_error](https://user-images.githubusercontent.com/48753146/178689667-b732077e-4552-461f-8737-d0a523b91776.PNG)
.

**Comment by WenmuZhou** [comment]
You can try this on the windows platform https://github.com/sdcb/PaddleSharp

**Reply by WENDGOUNDI** [reply]
Thank you @WenmuZhou. Any hint for python since I am currently using python while PaddleSharp is for c# according to the readings.

**Reply by Evezerest** [reply]
hi, pyinstaller does not adapt to PaddlePaddle, so we recommend you use https://github.com/QPT-Family/QPT to pack paddle applications. A translator is also needed for its Chinese README.

**Comment by Heryk13** [comment]
https://github.com/PaddlePaddle/PaddleOCR/discussions/11490
here is some tutorial i write for how to make a exe with paddleOCR

**Comment by Suvoo** [comment]
perfectly working solution (August 2024) : https://github.com/PaddlePaddle/PaddleOCR/discussions/11342

Tl,DR :
`pyinstaller main.py --collect-all paddleocr --collect-all pyclipper --collect-all imghdr --collect-all skimage --collect-all imgaug --collect-all scipy.io --collect-all lmdb`

**Comment by alex-dinh** [comment]
This works as of July 2025:

https://github.com/PaddlePaddle/PaddleOCR/issues/15918#issuecomment-3023825346

```
 pyinstaller test.py \
 	--collect-data paddlex \
 	--copy-metadata ftfy \
 	--copy-metadata imagesize \
 	--copy-metadata lxml \
 	--copy-metadata opencv-contrib-python \
 	--copy-metadata openpyxl \
 	--copy-metadata premailer \
 	--copy-metadata pyclipper \
 	--copy-metadata pypdfium2 \
 	--copy-metadata scikit-learn \
 	--copy-metadata shapely \
 	--copy-metadata tokenizers \
 	--copy-metadata einops \
 	--copy-metadata jinja2 \
 	--copy-metadata regex \
 	--copy-metadata tiktoken \
 	--add-binary "Path/to/your/site-packages/paddle/libs;."  \
 	--hidden-import "scipy._cyutility"

```
